### PR TITLE
:memo: Updating Tutorial-09, Tutorial-01 and Recursion page to resolve the issue #739 , #735 , #736 . Thanks.

### PR DESCRIPTION
**Pull request by LuffySama-Dev**
Solution for issue #739 

recurrsionTypeIssue01
recurrsionTypeIssue02

In above two images, if you will see we are getting the type error in method as the type of publicInput is not matching to the state.

So, I am just modifying the publicInputType to publicInput similar to the previous versions. You can check the similar fix in previous commit here:  #738

### CODE REVIEW

**Comment by Saurabhpatil-dev** [comment]
Hi @barriebyron ,

Sorry for adding multiple issue in one pull request. 
But these are all working. 
Let me know how I can help to keep them clean next time ?

Issue which are also fixed:
Tutorial runner work locally #736
main.ts in Tutorial 1 fails CI, tutorial runner #735

File: docs/zkapps/tutorials/01-hello-world.mdx [hunk 1]
```diff
@@ -379,8 +379,8 @@ All smart contracts that you create with the zkApp CLI use similar code:
 Comments break down each stage:
 
 ```ts src/main.ts
-19 const { privateKey: deployerKey, publicKey: deployerAccount } = Local.testAccounts[0];
-20 const { privateKey: senderKey, publicKey: senderAccount } = Local.testAccounts[1];
+19
```
Comment (barriebyron) [inline docs/zkapps/tutorials/01-hello-world.mdx:382]: @Saurabhpatil-dev  
To verify this code update, please make the corresponding change in the example code file https://github.com/o1-labs/docs2/blob/main/examples/zkapps/01-hello-world/src/main.ts

Comment (Saurabhpatil-dev) [reply docs/zkapps/tutorials/01-hello-world.mdx:382]: Hi @barriebyron ,

I compared both of the files and it seems the one which are asking me to update is already correct and matching to the changes I committed.

I am attaching the screenshot for your reference:

The file on **_left side is the one I have committed_** and the **_another one which you are asking me to update is on right side_**.  
The duplicated code was removed by me from the left file on line number 19-20. 
But when checked on the file in right side the duplicate code is already removed. So, I don't think so we need to update it. 
Please let me know if you would like me to update the file to match the line numbers in both files or make any other changes.

Screenshot 2023-11-22 at 10 32 27 PM

Thank you 😃

**Review by barriebyron — APPROVED** [review]
thanks @Saurabhpatil-dev for these fixes!

**Pull request merged**